### ISSUE TITLE
Will TimePicker support locale representations?

### ISSUE THREAD

**Issue by LunaFidalgo** [palantir/blueprint#6450]
#### Environment

- __Package version(s)__: `@blueprint/datetime: v5.2.2`
- __Browser and OS versions__: Chrome Version 117.0.5938.149 + macOs Ventura 13.5.2

#### Question
Hi, we are using the `TimePicker` component. We want to show the time in UTC, but it is not possible (without doing some hacky transformation) because internally, [the component uses these functions](https://github.com/palantir/blueprint/blob/develop/packages/datetime/src/components/time-picker/timePicker.tsx#L291-L294): `getHours, getMinutes, getSeconds`. All of them are representations using the local time. 

 For example, we have this: `"2023-09-09T09:43:06.206Z"`, and we want to represent the UTC time, but it is impossible because `TimePicker` will always show the local time representation. 

I would like to ask two things: 
1. Is this correct? Or am I missing something?
2. I have seen that the date time package is being improved to support local times. Is `TimePicker` going to be improved with this functionality?

Thank you very much!

**Our hack**
Using the functions available in [timeZoneUtils](https://github.com/palantir/blueprint/blob/develop/packages/datetime/src/common/timezoneUtils.ts), we managed to transform the date like this: 

```js
const dateToTransform = "2023-09-09T09:43:06.206Z"
const transformedDate = TimezoneUtils.convertLocalDateToTimezoneTime(
    new Date(dateToTransform),
    "Etc/UTC
);

console.log(transformedDate.toISOString()); // "2023-09-09T07:43:06.206Z" 
```

We also needed to transform it with `convertDateToLocalEquivalentOfTimezoneTime` in the `onChange` method.

**Comment by adidahiya** [comment]
Hi @LunaFidalgo, thanks for the question / feature request. We do want to improve the __TimePicker__ component to be more ergonomic when dealing with different timezones. I haven't looked into this closely yet, but I think this could be implemented with a `timezone?: string` prop similar to the __DateInput__ component. That prop value could be used to automatically call `TimezoneUtils.convertLocalDateToTimezoneTime()` on the incoming `value` prop and on the updated time during the `onChange` callback.

I hope to implement this FR in Blueprint v5.x, but also open to PRs if anyone wants to contribute.

**Comment by LunaFidalgo** [comment]
Hi @adidahiya, thank you for your answer. I would love to contribute with a PR! 

I was wondering if we could also add a prop to `DatePicker3` to use the `TimePicker` `timezone` prop you suggest. Otherwise, we will have the same issue while using `DatePicker`. 

Maybe something like `timePickerProps`?

**EDIT:** It wouldn't be that simple. We also need to transform the value in the `DatePicker` in case the day changes because of the timezone. Example: 09/09/2023, 23:43 => timeZone: "Europe/Madrid" => 10/09/2023, 01:43

**Comment by adidahiya** [comment]
> It wouldn't be that simple. We also need to transform the value in the `DatePicker` in case the day changes because of the timezone.

Yeah, there are certainly going to be some interesting edge cases to work through when implementing this feature. It also prompts the question of whether we should change the `value` type of __DatePicker__ from `Date` to `string` like we did for `DateInput` in v5.0 (see [datetime2 component migration](https://github.com/palantir/blueprint/wiki/datetime2-component-migration)) to allow the component to report the timezone accurately.
